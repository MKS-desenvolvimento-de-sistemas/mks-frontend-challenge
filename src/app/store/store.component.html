<div class="store">
    <h1>Store</h1>
    <div *ngIf="isLoading">Loading...</div>
    <app-product-card *ngFor="let product of products" [product]="product" (addToCartEvent)="addToCart($event)"></app-product-card>
  </div>
  