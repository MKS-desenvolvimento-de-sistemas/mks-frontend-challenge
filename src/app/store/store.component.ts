import { Component, OnInit } from '@angular/core';
import { ApiService, Product } from '..//services/api.service';

@Component({
  selector: 'app-store',
  templateUrl: './store.component.html',
  styleUrls: ['./store.component.scss']
})
export class StoreComponent implements OnInit {
  products: Product[] = [];
  isLoading = true;

  constructor(private apiService: ApiService) {}

  ngOnInit() {
    this.apiService.fetchProducts().then(products => {
      this.products = products;
      this.isLoading = false;
    });
  }

  addToCart(product: Product) {
    // Implementar a l√≥gica de adicionar ao carrinho
    console.log('Adicionar ao carrinho:', product);
  }
}
