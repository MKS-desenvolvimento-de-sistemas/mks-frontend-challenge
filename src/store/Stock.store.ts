import {createSlice} from '@reduxjs/toolkit';

const carrinho = createSlice({
	name: 'carrinho',
	initialState: {
		counter: 0,
		cartOpen: 'none',
	},
	reducers: {
		increment(state) {
			state.counter += 1;
		},
		decrement(state) {
			state.counter -= 1;
		},
		abrirCarrinho(state) {
			state.cartOpen = 'flex';
		},
		fecharCarrinho(state) {
			state.cartOpen = 'none';
		},
	},
});

export const {
	decrement,
	increment,
	abrirCarrinho,
	fecharCarrinho,
} = carrinho.actions;
export default carrinho.reducer;
