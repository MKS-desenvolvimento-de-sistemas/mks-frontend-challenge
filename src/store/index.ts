import { createSlice } from "@reduxjs/toolkit";
import { configureStore } from "@reduxjs/toolkit";
import { ICartItem } from "../pages/Cart";
const initialState = { items: [] } as any;

const cartSlice = createSlice({
  name: "carrinho",
  initialState,
  reducers: {
    addToCart(state, action) {
      state.items.push(action.payload);
    },

    addAdditional(state, action) {
      if (!state.items.some((item) => item.id === action.payload.id)) return;
      const item = state.items.find((item) => item.id === action.payload.id);

      state.items.push(item);
    },
    decreaseItem(state, action) {
      const remainingItems = state.items.filter(
        (item) => item.id === action.payload.id
      );

      if (remainingItems.length === 1) return;

      const itemIndex = state.items.findIndex(
        (el) => el.id === action.payload.id
      );

      state.items.splice(itemIndex, 1);
    },
    deleteItem(state, action) {
      const arr = state.items.filter((item) => item.id !== action.payload.id);
      state.items = arr;
      // do {
      //   const itemIndex = state.items.findIndex(
      //     (el) => el.id === action.payload.id
      //   );

      //   state.items.splice(itemIndex, 1);
      // } while (state.items.some((item) => item.id === action.payload.id));
    },
    deleteAllItems(state) {
      state.items = [];
    },
  },
});

const store = configureStore({
  reducer: cartSlice.reducer,
});

export const cartActions = cartSlice.actions;

export default store;
