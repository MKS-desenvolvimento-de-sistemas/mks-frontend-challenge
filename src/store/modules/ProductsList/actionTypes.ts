export const ALL_PRODUCTS = "@PRODUCTS/ALL";
