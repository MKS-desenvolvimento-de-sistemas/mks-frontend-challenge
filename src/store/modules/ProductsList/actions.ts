import { ALL_PRODUCTS } from "./actionTypes";

export const allProd = (products: any) => ({
  type: ALL_PRODUCTS,
  products,
});
