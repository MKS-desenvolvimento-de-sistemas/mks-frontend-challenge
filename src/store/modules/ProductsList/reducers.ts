import { ALL } from "dns";
import { ALL_PRODUCTS } from "./actionTypes";

export const reducerProducts = (state = [], action: { type: any }) => {
  switch (action.type) {
    case ALL_PRODUCTS:
      const { products }: any = action;
      return [...state, products];

    default:
      return state;
  }
};
