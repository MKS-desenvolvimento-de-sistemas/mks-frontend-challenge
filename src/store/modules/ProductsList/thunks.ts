import { allProd } from "./actions";
import { api } from "../../../services/Api";

export const allProdThunk =
  (setIsLoading: (arg0: boolean) => void) => (dispatch: any) => {
    setIsLoading(true);
    api
      .get("/products?page=1&rows=10&sortBy=id&orderBy=ASC")
      .then((response) => {
        dispatch(allProd(response.data.products));
        setIsLoading(false);
      })
      .catch((error) => {
        setIsLoading(false);
      });
  };
