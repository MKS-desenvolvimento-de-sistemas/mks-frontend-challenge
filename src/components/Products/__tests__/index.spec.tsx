import "@testing-library/jest-dom";
import { fireEvent, render, screen } from "@testing-library/react";
import ProductsList from "../";
import renderer from "react-test-renderer";
import { Provider } from "react-redux";
import store from "../../../redux/";
import { IProduct } from "../../../types/api";

const productCartExampleI: IProduct = {
  id: 1,
  name: "EXEMPLO 1",
  brand: "Exemplo 1",
  description: "Exemplo 1",
  price: "200",
  photo: "/vercel.svg",
  updatedAt: "2022",
  createdAt: "2022",
};

const productCartExampleII: IProduct = {
  id: 2,
  name: "EXEMPLO 2",
  brand: "Exemplo 2",
  description: "Exemplo 2",
  price: "350",
  photo: "/vercel.svg",
  updatedAt: "2022",
  createdAt: "2022",
};

const productsExample = [productCartExampleI, productCartExampleII];

describe("ProductsList Component", () => {
  it("should render the products list component correctly", () => {
    render(
      <Provider store={store}>
        <ProductsList products={productsExample} />
      </Provider>
    );

    const productsList = screen.getByTestId("products-list");

    expect(productsList).toHaveTextContent("EXEMPLO 1");
    expect(productsList).toHaveTextContent("EXEMPLO 2");
  });

  it("should match the snapshot", () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <ProductsList products={productsExample} />
        </Provider>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
