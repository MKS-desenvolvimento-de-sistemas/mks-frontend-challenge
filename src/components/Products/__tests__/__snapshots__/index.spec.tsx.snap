// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductsList Component should match the snapshot 1`] = `
<ul
  className="sc-bczRLJ kicoiy"
  data-testid="products-list"
>
  <li
    className="sc-gsnTZi jfSWcW"
    data-testid="card"
  >
    <div
      className="sc-dkzDqf dsBQjw"
    >
      <div
        className="sc-hKMtZM fNiXDV"
        data-testid="card--image"
      />
      <div
        className="sc-eCYdqJ jIhmma"
      >
        <h4>
          EXEMPLO 1
        </h4>
        <div
          className="sc-papXJ jHezEv"
          data-testid="price"
          type="product"
        >
          R$200
        </div>
        <p>
          Exemplo 1
        </p>
      </div>
    </div>
    <button
      className="sc-idiyUo eimsFO"
      data-testid="buy-button"
      onClick={[Function]}
      title="Adicionar ao Carrinho"
      type="button"
    >
      <svg
        fill="none"
        height="16px"
        stroke="currentColor"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="2"
        style={
          {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="16px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M6 2L3 6v14a2 2 0 0 0 2 2h14a2 2 0 0 0 2-2V6l-3-4z"
        />
        <line
          x1="3"
          x2="21"
          y1="6"
          y2="6"
        />
        <path
          d="M16 10a4 4 0 0 1-8 0"
        />
      </svg>
      Comprar
    </button>
  </li>
  <li
    className="sc-gsnTZi jfSWcW"
    data-testid="card"
  >
    <div
      className="sc-dkzDqf dsBQjw"
    >
      <div
        className="sc-hKMtZM fNiXDV"
        data-testid="card--image"
      />
      <div
        className="sc-eCYdqJ jIhmma"
      >
        <h4>
          EXEMPLO 2
        </h4>
        <div
          className="sc-papXJ jHezEv"
          data-testid="price"
          type="product"
        >
          R$350
        </div>
        <p>
          Exemplo 2
        </p>
      </div>
    </div>
    <button
      className="sc-idiyUo eimsFO"
      data-testid="buy-button"
      onClick={[Function]}
      title="Adicionar ao Carrinho"
      type="button"
    >
      <svg
        fill="none"
        height="16px"
        stroke="currentColor"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="2"
        style={
          {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="16px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M6 2L3 6v14a2 2 0 0 0 2 2h14a2 2 0 0 0 2-2V6l-3-4z"
        />
        <line
          x1="3"
          x2="21"
          y1="6"
          y2="6"
        />
        <path
          d="M16 10a4 4 0 0 1-8 0"
        />
      </svg>
      Comprar
    </button>
  </li>
</ul>
`;
